### DISCUSSION TITLE
I cant use copilot

### DISCUSSION THREAD

**Discussion by ibrohim2911** [community/community#8210] [Copilot Conversations]
GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 140.82.121.5:443"
why it show me 
I signed in to github from vs code and github copilot

**Comment by davecheney** [comment]
Hello, I'm sorry you are having difficulties. It looks like something is interfering with your connection to api.github.com. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by ibrohim2911** [reply]
no I dont use any vpn or others. if I come to api.github.io there has codes like json

**Reply by ibrohim2911** [reply]
thanks for your reply

**Comment by persistz** [comment]
+1 same question, after a day of pleasant experience, I could no longer connect to it

**Reply by davecheney** [reply]
Can you please past the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by v587su** [reply]
I have the same problem, and this is my output:
~~~ 
 [ERROR] [default] [2021-11-30T02:30:54.611Z] GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 140.82.114.3:443"
~~~

**Reply by davecheney** [reply]
Please see my question above, https://github.com/github/feedback/discussions/8210#discussioncomment-1715193

**Reply by v587su** [reply]
> Please see my question above, [#8210 (comment)](https://github.com/github/feedback/discussions/8210#discussioncomment-1715193)

I have closed my Microsoft Defender and VPN. It still doesn't work. But I can successfully ping 140.82.121.5 (the IP shown in the log of ibrohim2911)

**Reply by davecheney** [reply]
Ping and https use different protocols, being able to ping an IP address is good, but it’s not the whole story.

**Reply by v587su** [reply]
> Ping and https use different protocols, being able to ping an IP address is good, but it’s not the whole story.

Yeah, but I can't ping the IP in my error message. I'm wondering whether there is a way to switch the server in my extension.

**Reply by persistz** [reply]
Hi bro, what a coincidence.
My server is 140.82.114.3 too.
I try "curl -vv https://140.82.114.3 --insecure", I can't get a reply unless I use a proxy
So I suspect that the cause of this problem may be that github copilot does not support proxy.

**Reply by davecheney** [reply]
Sadly at the moment copilot does not support proxies

**Reply by persistz** [reply]
> Sadly at the moment copilot does not support proxies

That's OK. We fixed it by changing hosts. As we can connect 140.82.121.5 without proxy, so what we need is just modify the ip.

**Reply by davecheney** [reply]
You’ll probably have to play some games with your /etc/hosts file
